modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -602,6 +651,47 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
 
         CV_Assert(outputs.size() == 1);
         const int nstripes = getNumThreads();
+
+        if (channelsModeInput == ELTWISE_CHANNNELS_SAME && inputs[0].dims > 2)
```

Comment:
Please add bailout code in `forward_ocl()` (line 546) for new functionality:

```.diff
         if ((inputs_.depth() == CV_16S && op != SUM) || (channelsMode != ELTWISE_CHANNNELS_SAME))
             return false;
 
+        if (hasVecInput)
+            return false;  // TODO not implemented yet: https://github.com/opencv/opencv/pull/19477
```

Reply:
@alalek, ``forward_ocl()`` was corrected.